### PR TITLE
Repeating fields bug

### PR DESCRIPTION
**Pull request by zeeshan-vd**
Cleared invalid repeating fields that have been deleted.

### CODE REVIEW

**Comment by vincent-karuri** [comment]
Shouldn't we look at https://github.com/OpenSRP/opensrp-client-native-form/blob/2b750a3abbc777b9a992f04a96137eb0f40cba35/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java#L116 to see why fields are not removed from `FormDataViews` during deletion?

**Comment by zeeshan-vd** [comment]
When the first time we try to submit the form with invalid repeating fields, `validateAndWriteValues` method puts those invalid fields to `invalidFields`. Now when we remove the repeating fields from the UI by putting zero as the count and submit the form, `validateAndWriteValues` will check each field and if it is valid it will remove it from `invalidFields` but since we have already cleared the repeating fields through the UI so they are not a part of `FormDataViews` but still a part of `invalidFields` so they don't get removed.

**Comment by vincent-karuri** [comment]
> When the first time we try to submit the form with invalid repeating fields, `validateAndWriteValues` method puts those invalid fields to `invalidFields`. Now when we remove the repeating fields from the UI by putting zero as the count and submit the form, `validateAndWriteValues` will check each field and if it is valid it will remove it from `invalidFields` but since we have already cleared the repeating fields through the UI so they are not a part of `FormDataViews` but still a part of `invalidFields` so they don't get removed and we see the validation error.

Cool, can we then remove them from invalid fields at the point where they're being deleted?

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java [hunk 1]
```diff
@@ -443,6 +443,16 @@ public static void makeFieldHidden(JSONObject fieldObject) {
         }
     }
 
+    public static void removeDeletedInvalidFields(String fieldKeyPrefix, Map<String, ValidationStatus> invalidFields, ArrayList<String> removeThisFields) {
+        for (String removeField : removeThisFields) {
```
Comment (vincent-karuri) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java:447]: Prefer nouns to verbs for variables e.g. `fieldToRemove` from `fieldsToRemove` as opposed to `removeField` and `removeTheseFields.`

**Comment by vincent-karuri** [comment]
@zeeshan-vd could you bump the lib version?

**Review by vincent-karuri — APPROVED** [review]

**Pull request merged**